openverse_catalog/dags/providers/provider_api_scripts/stocksnap.py

```diff
@@ -28,175 +28,165 @@
 DELAY = 1  # in seconds
 RETRIES = 3
 HOST = "stocksnap.io"
-ENDPOINT = f"https://{HOST}/api/load-photos/date/desc"
+ENDPOINT_BASE = f"https://{HOST}/api/load-photos/date/desc"
 IMAGE_CDN = "https://cdn.stocksnap.io/img-thumbs/960w"
 PROVIDER = prov.STOCKSNAP_DEFAULT_PROVIDER
-HEADERS = {
-    "Accept": "application/json",
-}
 
-delayed_requester = DelayedRequester(DELAY)
-image_store = ImageStore(provider=PROVIDER)
 
-license_url = "https://creativecommons.org/publicdomain/zero/1.0/"
-license_info = get_license_info(license_url=license_url)
+class StockSnapDataIngester(ProviderDataIngester):
+    providers = {"image": prov.STOCKSNAP_DEFAULT_PROVIDER}
+    batch_limit = 1000
+    delay = 1  # in seconds
+    headers = {
+        "Accept": "application/json",
+    }
 
+    def __init__(self):
+        super(StockSnapDataIngester, self).__init__()
+        self._page_counter = 1
 
-def main():
-    """
-    This script pulls all the data from the StockSnap and writes it into a
-    .TSV file to be eventually read into our DB.
-    """
-
-    logger.info("Begin: StockSnap script")
-    image_count = _get_items()
-    image_store.commit()
-    logger.info(f"Total images pulled: {image_count}")
-    logger.info("Terminated!")
-
-
-def _get_items():
-    item_count = 0
-    page_number = 1
-    should_continue = True
-    while should_continue:
-        page_endpoint = f"{ENDPOINT}/{page_number}"
-        batch_data = _get_batch_json(endpoint=page_endpoint)
-        if isinstance(batch_data, list) and len(batch_data) > 0:
-            item_count = _process_item_batch(batch_data)
-            page_number += 1
-        else:
-            should_continue = False
-    return item_count
-
-
-def _get_batch_json(
-    endpoint=ENDPOINT, headers=None, retries=RETRIES, query_params=None
-):
-    if headers is None:
-        headers = HEADERS.copy()
-    response_json = delayed_requester.get_response_json(
-        endpoint, retries, query_params, headers=headers
-    )
-    if response_json is None:
+    def get_next_query_params(self):
```

Comment:
This will throw an error when `ingest_records` attempts to invoke it with `old_query_params`
```suggestion
    def get_next_query_params(self, old_query_params, **kwargs):
```